From f3544123dc7f03be33cd829efb647fe6da51b71a Mon Sep 17 00:00:00 2001
From: Daniel Chapin <chapin@google.com>
Date: Thu, 22 Aug 2024 17:56:59 +0000
Subject: [PATCH 04/18] Revert "flatland: ConsumerBase-based classes now create
 their own BufferQueues"

This reverts commit a28188817bffb7a9299e74f679893d71199ef4e3.

Reason for revert: This CL has been found via bisection to be the possible root cause of the DF blocking bug b/360771012 A revert will be made and staged to validate the bisection result.
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:af6a1a8ff83fa7fe0e1546c98e4fe46962080462)
Merged-In: Id4b41741736ac07b145edebae9f03ac359915fca
Change-Id: Id4b41741736ac07b145edebae9f03ac359915fca
---
 cmds/flatland/GLHelper.cpp | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/cmds/flatland/GLHelper.cpp b/cmds/flatland/GLHelper.cpp
index 77e732805e..c163095c50 100644
--- a/cmds/flatland/GLHelper.cpp
+++ b/cmds/flatland/GLHelper.cpp
@@ -18,7 +18,6 @@
 
 #include <GLES2/gl2.h>
 #include <GLES2/gl2ext.h>
-#include <com_android_graphics_libgui_flags.h>
 #include <gui/SurfaceComposerClient.h>
 #include <ui/DisplayMode.h>
 
@@ -203,14 +202,6 @@ bool GLHelper::getShaderProgram(const char* name, GLuint* outPgm) {
 
 bool GLHelper::createNamedSurfaceTexture(GLuint name, uint32_t w, uint32_t h,
         sp<GLConsumer>* glConsumer, EGLSurface* surface) {
-#if COM_ANDROID_GRAPHICS_LIBGUI_FLAGS(WB_CONSUMER_BASE_OWNS_BQ)
-    sp<GLConsumer> glc = new GLConsumer(name, GL_TEXTURE_EXTERNAL_OES, false, true);
-    glc->setDefaultBufferSize(w, h);
-    glc->getSurface()->setMaxDequeuedBufferCount(2);
-    glc->setConsumerUsageBits(GRALLOC_USAGE_HW_COMPOSER);
-
-    sp<ANativeWindow> anw = glc->getSurface();
-#else
     sp<IGraphicBufferProducer> producer;
     sp<IGraphicBufferConsumer> consumer;
     BufferQueue::createBufferQueue(&producer, &consumer);
@@ -221,7 +212,6 @@ bool GLHelper::createNamedSurfaceTexture(GLuint name, uint32_t w, uint32_t h,
     glc->setConsumerUsageBits(GRALLOC_USAGE_HW_COMPOSER);
 
     sp<ANativeWindow> anw = new Surface(producer);
-#endif // COM_ANDROID_GRAPHICS_LIBGUI_FLAGS(WB_CONSUMER_BASE_OWNS_BQ)
     EGLSurface s = eglCreateWindowSurface(mDisplay, mConfig, anw.get(), nullptr);
     if (s == EGL_NO_SURFACE) {
         fprintf(stderr, "eglCreateWindowSurface error: %#x\n", eglGetError());
-- 
2.43.0

